### DISCUSSION TITLE
How to estimate gas of a receive method() in truffle

### DISCUSSION THREAD

**Discussion by SD170** [ConsenSys-archive/truffle#5353] [General Q&A]
I know that if my contract has a name() method. I can go to the **truffle console** and try

> contractInstance.name.estimateGas()

and this will give the estimate gas for that method.

but what if have a receive method inside my contract to take ether:

```
receive() external payable {
} 
```
and I wanna know it's gas. Then what is the correct syntax?

I've tried:

> contractInstance.estimateGas()

as I thought as receive is the default function for ether payment. But didn't work.

> contractInstance.receive.estimateGas()

this doesn't work either.

can I get some help?

**Comment by haltman-at** [comment]
Hi!  Unfortunately, there's no convenient way to do this at the moment.  Doing `instance.estimateGas({})` would be a good way to do this if we supported it, but unfortunately at the moment we don't... I've put up an issue for this, #5374, and hopefully we can get that done soon.

In the meantime, you can do this manually using web3.js.  If you have your contract instance, you can do
```
web3.eth.estimateGas({ from: accounts[0], to: instance.address })
```
in order to perform the gas estimation that you want.  You can of course also include a `value` parameter if you want to send some number of wei, in case that affects things.

Hope that's helpful!

**Reply by SD170** [reply]
Hi, @haltman-at thanks for the reply. The thing is, I've been using truffle for some time. And I was wondering if I could contribute some. If it's okay I can try adding this feature (instance.estimateGas() for the fallback/receive methods), If I get some guide. 

Should I go to the issue you've created and make this comment there? or I can try to add the feature and do a PR directly?
I know you're a maintainer of this project, and I'm kinda new to open source, so I thought of asking you.

**Reply by haltman-at** [reply]
You could make a PR, but I have to warn you, the code for `@truffle/contract` is quite hairy...

**Reply by SD170** [reply]
Thanks for the heads up :) I'll try. Otherwise, I'll try to contribute with something easier.

**Reply by haltman-at** [reply]
FWIW, I've put up PR #5426 to address this!  Hopefully it gets merged and released soon.

**Reply by SD170** [reply]
Great man. I saw the code. I also hope it gets merged soon :)
